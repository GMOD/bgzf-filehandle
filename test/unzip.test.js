const fs = require('fs')
const { pakoUnzip, nodeUnzip, unzipChunk } = require('../src/unzip')

describe('unzip', () => {
  it('can unzip bgzip-1.txt.gz', async () => {
    const testData = fs.readFileSync(require.resolve('./data/bgzip-1.txt.gz'))
    const fromPako = await pakoUnzip(testData)
    const fromNode = await nodeUnzip(testData)
    expect(fromNode).toEqual(fromPako)
    expect(fromNode.length).toEqual(65569)
    expect(fromPako.length).toEqual(65569)
  })
})

// describe('unzip', () => {
//   it('can unzip bgzip-1.txt.gz', async () => {
//     const testData = fs.readFileSync(require.resolve('./data/paired.bam'))
//     const { dpositions, cpositions } = await unzipChunk(testData)
//     expect(dpositions).toMatchSnapshot()
//     expect(cpositions).toMatchSnapshot()
//   })
// })
