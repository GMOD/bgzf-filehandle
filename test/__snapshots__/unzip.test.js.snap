// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unzip can unzip bgzip-1.txt.gz 1`] = `
Array [
  0,
  22579,
  87646,
  152675,
  217701,
  282681,
  287762,
]
`;

exports[`unzip can unzip bgzip-1.txt.gz 2`] = `
Array [
  0,
  4682,
  24403,
  43638,
  62447,
  82222,
  84033,
]
`;
